UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.FulfilmentService/Services/FulfilmentDataService.cs

```diff
@@ -43,34 +45,54 @@ public class FulfilmentDataService : IFulfilmentDataService
             this.fulfilmentAncillaryFiles = fulfilmentAncillaryFiles;
             this.fulfilmentSalesCatalogueService = fulfilmentSalesCatalogueService;
             this.fulfilmentCallBackService = fulfilmentCallBackService;
+            this.monitorHelper = monitorHelper;
         }
 
         public async Task<string> CreateExchangeSet(SalesCatalogueServiceResponseQueueMessage message)
         {
+            DateTime createExchangeSetTaskStartAt = DateTime.UtcNow;
             string homeDirectoryPath = configuration["HOME"];
             var exchangeSetPath = Path.Combine(homeDirectoryPath, DateTime.UtcNow.ToString("ddMMMyyyy"), message.BatchId, fileShareServiceConfig.Value.ExchangeSetFileFolder);
             var exchangeSetRootPath = Path.Combine(exchangeSetPath, fileShareServiceConfig.Value.EncRoot);
             var exchangeSetPathForUploadZipFile = Path.Combine(homeDirectoryPath, DateTime.UtcNow.ToString("ddMMMyyyy"), message.BatchId);
             var listFulfilmentData = new List<FulfilmentDataResponse>();
-
+            DateTime downloadEncFilesFromFileShareServiceTaskStartAt = DateTime.UtcNow;
```

Comment:
This line of code has to be inside of if statement at line 61 and also change variable name downloadEncFilesFromFileShareServiceTaskStartAt to queryAndDownloadEncFilesFromFileShareServiceTaskStartedAt

Reply:
Done